[Toh Jia Han] iP

tohjh opened the pull request:
#My IP Project: Chobo
>you have better things to do ~Jia Han

## Introduction
My iP (individual project) is a simple command-line task manager application that helps you keep track of your tasks, deadlines, and events. It's built to be:
- **User-friendly**
- **Efficient**
- **Customizable**

## How to Get Started
1. Download the latest release from [here](https://github.com/tohjh/ip/releases/tag/v0.1).
2. Run the `Chobo.jar` file.

## Features
- [x] Manage tasks
- [ ] Manage deadlines (coming soon)
- [ ] Custom themes (coming soon)

## Feedback
:tada: Let me know if there are any feedbacks! Excited to see your feedbacks!

## Code Example
Here’s a simple example of the main method in Java:

```java
public class Chobo {
    public static void main(String[] args) {
        new Chobo().run() ;
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tohjh committed:
add Level 0

tohjh committed:
added Level 1

tohjh committed:
added level 2

tohjh committed:
Added level 4

tohjh committed:
actual level 4

tohjh committed:
Revert "actual level 4"

This reverts commit c5d2d19d770b37705a29b7073e389d17d003e333.

tohjh committed:
Reapply "actual level 4"

This reverts commit 2839012c714aaed40d645bb68a26bf7d1a877412.

tohjh committed:
Tried Automated UI testing

tohjh committed:
Add exception handling

tohjh committed:
Add delete function

tohjh committed:
Add more exceptions

tohjh committed:
Add javadoc comments

tohjh committed:
format

tohjh committed:
format

tohjh committed:
Add save function

tohjh committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Task.java

tohjh committed:
Add date time formatting

tohjh committed:
Update level 8

tohjh committed:
Revert "Update level 8"

This reverts commit 3622095ae310b8cae901f09e036fab87c1aa46fe.

tohjh committed:
Update level 8

tohjh committed:
Merge branch 'branch-Level-8'

tohjh committed:
Use more OOP

tohjh committed:
Organize into packages

tohjh committed:
Merge branch 'add-gradle-support'

tohjh committed:
Add JUnit Tests

tohjh committed:
Add jar

tohjh committed:
Add Javadocs

tohjh committed:
Test commit

tohjh committed:
Revert "Test commit"

This reverts commit 21828973048e55ff3deb66f74037d474376c52c2.

tohjh committed:
Add JavaDocs

tohjh committed:
Merge branch 'branch-A-JavaDoc'

tohjh committed:
Add coding standards

tohjh committed:
Merge branch 'branch-A-CodingStandard'

tohjh committed:
Add find

tohjh committed:
Merge branch 'branch-Level-9'

tohjh committed:
Add some GUI

hooyushing commented on src/main/java/chobo/Parser.java:
```diff
@@ -0,0 +1,119 @@
+     * @throws InputException the input exception
+     */
+    public static boolean parse(String fullCommand, TaskList taskList, Ui ui, Storage storage) throws InputException {
+        String[] input = fullCommand.split(" ", 2);
```
Is it possible that the name is changed to a plural?

hooyushing reviewed (commented):
Hello! Good job on naming! The conventions match what are required except for one of them. :D

tohjh committed:
Add functional chatbot GUI

tohjh committed:
Add level 10

tohjh committed:
Add level 10 update

tohjh committed:
Add updates

tohjh committed:
Merge branch 'branch-Level-10'

tsulemornorfi commented on src/main/java/chobo/Main.java:
```diff
@@ -0,0 +1,37 @@
+    /**
+     * The entry point of application.
+     *
+     * @param args the input arguments
```
there should be a period after "arguments"

currynia reviewed (commented):
LGTM. Just some fixes needed for headers.

mrnobody0505 commented on src/main/java/chobo/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+        super(name, done);
+        this.unformattedDate = date;
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("d-M-yyyy HHmm");
+        this.by = LocalDateTime.parse(date.trim(), dateTimeFormatter);
```
Consider using name deadline instead of by for deadline date so that it might be more understandable

mrnobody0505 commented on src/main/java/chobo/Event.java:
```diff
@@ -0,0 +1,52 @@
+    private String unformattedTo;
+    private LocalDateTime from;
+    private LocalDateTime to;
+    /**
```
Consider changing fields name from/to to startDate/endDate

tohjh committed:
Add assertions to validate assumptions

Added assertions help to validate key assumptions in code. Assertions
added includes checks for non null values and correct logic at different
points of the code. Assertions intended to catch potential issues early.

tohjh committed:
Merge pull request #1 from tohjh/branch-A-Assertions

Add assertions to validate assumptions

tohjh committed:
Merge branch 'master' of https://github.com/tohjh/ip

tohjh committed:
Refactor code

Examined the existing codebase and applied necessary refactoring to
improve code quality, ensuring a structured and maintainable codebase.
Specific improvements include better naming conventions, removal of
excess methods, and more consistent formatting.

tohjh committed:
Merge branch 'master' into branch-A-CodeQuality

tohjh committed:
Revert "Merge branch 'master' into branch-A-CodeQuality"

This reverts commit 46de7fd58e69c7f3e4dd1a28881976cdb1020fbb, reversing
changes made to cdda5040714dc283d66327d5e7fcbe8cf7b75e8c.

tohjh committed:
Merge pull request #2 from tohjh/branch-A-CodeQuality

Refactor code

tohjh committed:
Merge branch 'master' of https://github.com/tohjh/ip

tohjh committed:
Add extension function

tohjh committed:
Merge branch 'branch-BCD-Extension'

tohjh committed:
Refactor code

tohjh committed:
Add user guide

tohjh committed:
Reformat user guide

tohjh committed:
Add javadoc

tohjh committed:
Rename variables

tohjh committed:
Update README.md
